src/vs/base/browser/ui/dropdown/dropdown.ts

```diff
@@ -248,19 +252,24 @@ export class DropdownMenu extends BaseDropdown {
 			getActionItem: action => this.menuOptions && this.menuOptions.actionItemProvider ? this.menuOptions.actionItemProvider(action) : null,
 			getKeyBinding: action => this.menuOptions && this.menuOptions.getKeyBinding ? this.menuOptions.getKeyBinding(action) : null,
 			getMenuClassName: () => this.menuClassName,
-			onHide: () => this.onHide(),
+			onHide: () => this.hideMenu.schedule(),
```

Comment:
Why is this suddenly async?

Reply:
the onhide calls all became async because this callback happens before we get the click event on the button in the native case. That means the button will get a click after the menu is not shown anymore and just reshow it which is exactly what we are trying to prevent. alternative proposal/solutions accepted.